cmd/kube-scheduler/app/options/options.go

```diff
@@ -186,7 +188,12 @@ func (o *Options) ApplyTo(c *schedulerappconfig.Config) error {
 			return err
 		}
 
-		// use the loaded config file only, with the exception of --address and --port. This means that
+		// flag feature gates take precedence over feature gates in config file
+		for name, value := range o.ComponentConfig.FeatureGates {
+			cfg.FeatureGates[name] = value
+		}
+
+		// use the loaded config file only, with the exception of --address, --port and --feature-gates. This means that
```

Comment:
This block of code refers to `--address` and `--port`. `--feature-gates` is handled above this comment. For readability, you could do `cliFeatureGates := o.ComponentConfig.FeatureGates` above and do the replacement below.

/hold
But let's hold for the discussion on the issue.